viadot/task_utils.py

```diff
@@ -37,9 +37,14 @@ def add_ingestion_metadata_task(
     Args:
         df (pd.DataFrame): input DataFrame.
     """
-    df2 = df.copy(deep=True)
-    df2["_viadot_downloaded_at_utc"] = datetime.now(timezone.utc).replace(microsecond=0)
-    return df2
+    if len(df.columns) == 0:
```

Comment:
Or just `if not df.columns` which is closer to natural language ("if there are no columns"). Also, please add a comment explaining why we do this check as it's not self-evident (ie. why we can't use df.empty here).

Reply:
`if not df.columns` causes an error so I went back to `if len(df.columns) == 0`

Reply:
Interesting, what's the error?